src/services/dealer-price/index.ts

```diff
@@ -0,0 +1,116 @@
+import util from "util"
+
+import { credentials } from "@grpc/grpc-js"
+
+import { UnknownDealerPriceServiceError } from "@domain/dealer-price"
+
+import { baseLogger } from "../logger"
+
+import { PriceServiceClient } from "./proto/services/price/v1/price_service_grpc_pb"
+import {
+  GetExchangeRateForImmediateUsdBuyRequest,
+  GetExchangeRateForImmediateUsdBuyResponse,
+  GetExchangeRateForImmediateUsdSellRequest,
+  GetExchangeRateForImmediateUsdSellResponse,
+  GetExchangeRateForFutureUsdBuyRequest,
+  GetExchangeRateForFutureUsdBuyResponse,
+  GetExchangeRateForFutureUsdSellRequest,
+  GetExchangeRateForFutureUsdSellResponse,
+} from "./proto/services/price/v1/price_service_pb"
+
+const serverPort = process.env.PRICE_SERVER_PORT ?? "50055"
+const client = new PriceServiceClient(
+  `localhost:${serverPort}`,
+  credentials.createInsecure(),
+)
+
+const clientGetExchangeRateForImmediateUsdBuy = util.promisify(
+  client.getExchangeRateForImmediateUsdBuy.bind(client),
+)
+const clientGetExchangeRateForImmediateUsdSell = util.promisify(
+  client.getExchangeRateForImmediateUsdSell.bind(client),
+)
+const clientGetExchangeRateForFutureUsdBuy = util.promisify(
+  client.getExchangeRateForFutureUsdBuy.bind(client),
+)
+const clientGetExchangeRateForFutureUsdSell = util.promisify(
+  client.getExchangeRateForFutureUsdSell.bind(client),
+)
+
+export const DealerPriceService = (): IDealerPriceService => {
+  const getExchangeRateForImmediateUsdBuy = async function (
+    amountInSatoshis: Satoshis,
+  ): Promise<UsdCents | DealerPriceServiceError> {
+    try {
+      const response = (await clientGetExchangeRateForImmediateUsdBuy(
+        new GetExchangeRateForImmediateUsdBuyRequest().setAmountInSatoshis(
+          amountInSatoshis,
+        ),
+      )) as GetExchangeRateForImmediateUsdBuyResponse
+      return response.getPriceInUsd() as UsdCents
+    } catch (error) {
+      baseLogger.error(
+        { error },
+        "GetExchangeRateForImmediateUsdBuy unable to fetch price",
+      )
+      return new UnknownDealerPriceServiceError()
```

Comment:
For `new Unknown...Error` we've so far been passing the caught error in for better visibility higher up in the code.
```suggestion
      return new UnknownDealerPriceServiceError(error)
```

Reply:
well spotted thank you for all the above!